frb_codegen/src/generator/rust/ty_primitive_list.rs

```diff
@@ -8,18 +8,29 @@ type_rust_generator_struct!(TypePrimitiveListGenerator, IrTypePrimitiveList);
 
 impl TypeRustGeneratorTrait for TypePrimitiveListGenerator<'_> {
     fn wire2api_body(&self) -> Option<String> {
-        Some(
-            "unsafe {
-            let wrap = support::box_from_leak_ptr(self);
-            support::vec_from_leak_ptr(wrap.ptr, wrap.len)
-        }"
+        Some(match &self.ir {
+            IrTypePrimitiveList::PrimitiveList { .. } => "unsafe {
+                    let wrap = support::box_from_leak_ptr(self);
+                    support::vec_from_leak_ptr(wrap.ptr, wrap.len)
+                }"
             .into(),
-        )
+            IrTypePrimitiveList::PrimitiveArray { len, .. } => format!(
+                "unsafe {{
+                    let wrap = support::box_from_leak_ptr(self);
+                    debug_assert_eq!(wrap.len, {len});
+                    let slice = core::slice::from_raw_parts_mut(wrap.ptr, wrap.len as usize);
+                    let ptr = slice.as_mut_ptr() as *mut {};
+                    let boxed = support::box_from_leak_ptr(ptr);
+                    *boxed
```

Comment:
TODO(fzyzcjy): If we use the delegate approach then it is ok; otherwise @fzyzcjy should carefully review these several unsafe code segments again

Reply:
Even with the use of `delegate`, the use of `unsafe` cannot be avoided and I really need someone to check that this part of `unsafe` is as expected

Reply:
May I know why unsafe cannot be avoided? Since if we delegate PrimitiveArray via a PrimitiveList, we have the following code:

* convert PrimitiveArray to a PrimitiveList - safe (possibly using some 3rd party crates, but definitely can be safe)
* use existing code of transfering PrimitiveList data between Rust and Dart - unsafe, but all unsafe are old code that exist for a long time, no new code at all now.